sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -842,12 +847,16 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     if (maybeSchemaFromTableProps.isDefined) {
       val schemaFromTableProps = maybeSchemaFromTableProps.get
       val partColumnNames = getPartitionColumnsFromTableProperties(table)
-      val reorderedSchema = reorderSchema(schema = schemaFromTableProps, partColumnNames)
```

Comment:
This is in the method `restoreHiveSerdeTable`, I think we should always reorder the columns to match the Hive behavior.